addons/uom/static/src/js/uom_field_widget.js

```diff
@@ -0,0 +1,151 @@
+                if (uom_precision < precision) {
+                    this.do_warn(_.str.sprintf(_t('Only %s decimals are allowed.</br> This setting can be configured on the Unit of Measure.'), uom_precision));
+                    this.$input.val(value.slice(0,-1));
+                    return;
+                }
+            }
+
+            const uom_rounding = this.formatOptions.uom && this.formatOptions.uom.rounding;
+            const rounded = utils.round_precision(value, uom_rounding);
+            if (! utils.float_is_zero(parseFloat(value) - rounded, 13)) {
+                this.do_warn(_.str.sprintf(_t('The rounding %s is not respected.</br> This setting can be configured on the Unit of Measure.'), uom_rounding));
+            }
+        },
+    });
+    field_registry.add('measure', FieldMeasure);
+
+    /**
+     * Returns a string representing a float.  The result takes into account the
+     * user settings (to display the correct decimal separator).
+     *
+     * @param {float|false} value the value that should be formatted
+     * @param {Object} [field] a description of the field (returned by fields_get
+     *   for example).  It may contain a description of the number of digits that
+     *   should be used.
+     * @param {Object} [options] additional options to override the values in the
+     *   python description of the field.
+     * @param {Object} [options.uom] the description of the uom to use
+     * @param {integer} [options.uom_id]
+     *        the id of the 'uom.uom' to use (ignored if options.uom)
+     * @param {string} [options.uom_field]
+     *        the name of the field whose value is the uom id
+     *        (ignore if options.uom or options.uom_id)
+     *        Note: if not given it will default to the field uom_field value
+     *        or to 'uom_id'.
+     * @param {Object} [options.data]
+     *        a mapping of field name to field value, required with
+     *        options.uom_field
+     * @returns {string}
+     */
+    fieldUtils.format.measure = function(value, field, options) {
+        options = options || {};
+        if (value === false) {
+            return "";
+        }
+        var uom = options.uom;
+        if (!uom) {
+            var uom_id = options.uom_id;
+            var uom_field = options.uom_field || field.uom_field || 'uom_id';
+            if (!uom_id && options.data) {
+                uom_id = options.data[uom_field] && options.data[uom_field].res_id;
+            } else if (!uom_id && options) {
+                uom_id = options[uom_field] && options[uom_field].res_id;
+            }
+            uom = session.uoms[uom_id];
+        }
+
+        var l10n = core._t.database.parameters;
+        var precision;
+        if (uom && uom.decimal_places) {
+            precision = uom.decimal_places;
+        } else {
+            precision = 0;
+        }
+        var formatted = _.str.sprintf('%.' + precision + 'f', value || 0).split('.');
```

Comment:
```suggestion
        let uom = options.uom;
        if (!uom) {
            let uom_id = options.uom_id;
            const uom_field = options.uom_field || field.uom_field || 'uom_id';
            if (!uom_id && options.data) {
                uom_id = options.data[uom_field] && options.data[uom_field].res_id;
            } else if (!uom_id && options[uom_field]) {
                uom_id = options[uom_field].res_id;
            }
            uom = session.uoms[uom_id];
        }

        const l10n = core._t.database.parameters;
        const precision = (uom && uom.decimal_places) || 0;
        const formatted = _.str.sprintf('%.' + precision + 'f', value || 0).split('.');
```
(`const` and `let` ftw :shipit: ) 

Also, an empty object (`{}`) is truthy, so I think it's better to test `options[uom_field]` instead of only `options` at the line 131.